### DISCUSSION TITLE
How do I upgrade my current model to large-v3?

### DISCUSSION THREAD

**Discussion by asyakovenko** [kaixxx/noScribe#77] [Q&A] [answered]
Hi. I compared the performance of large-v2 and large-v3 using whisper.cpp . According to subjective feelings, it seems that large-v3 gives a better result. But I couldn't figure out how to update or replace the large-v2 model in the Mac OS version with large-v3. 

I looked at the instructions for faster whisper on Github https://github.com/SYSTRAN/faster-whisper , on HF searched in https://huggingface.co/distil-whisper/distil-large-v3

Please help me!

**Comment by gernophil** [comment]
I am not sure, you can. macOS aren't designed to be very "open". You could try to search the folder `noScribe.app/Contents/Resources/models/faster-whisper-large-v2` or `noScribe.app/Contents/Resources/models/faster-whisper-small` and replace one of its contents with the new model. Don't rename the folders. because noScribe will be looking for these two folders only.

We should maybe update to this new version on our site, @kaixxx?

**Comment by kaixxx** [comment, marked as answer]
I've also read quite a view negative reviews about v3 of the large model, e.g. that hallucination is worse. In my own test, I couldn't find a clear winner between v2 & 3. 

You can try the suggestion of @gernophil. Use this model: https://huggingface.co/Systran/faster-whisper-large-v3/tree/main Unfortunatly, it's propably not going to work with the version of the faster-whisper library we use. You can update this too, if you install noScribe from source. But i'm not sure if it is worth the hassle. 

I will consider switching to v3 in the future, but don't have the time for that atm.

**Reply by gernophil** [reply]
@kaixxx, maybe we can add a "slot" in the models selection for v3, so people can add it manually, if they want to.

**Reply by kaixxx** [reply]
> maybe we can add a "slot" in the models selection

Yes, but I'm thinking about a more general approach how people could add more models to noScribe if they want. There are also some finetuned versions of whisper for certain languages out there. 

I would suggest a folder structure like this: 
```
models
  |__ whisper
    |__ fast
    |__ precise
    |__ precise_v3
    |__ fast_english
    ... (users can add their own models)
```

The names of the subfolders (fast, precise, precise_v3...) would then appear in the GUI in the dropdown menu for "Quality" (or a renamed option that replaces this). 

Problem is, we also need to update faster-whisper for this, which has consequences for other libraries as well as the pyinstaller scripts. This needs proper testing.

**Reply by asyakovenko** [reply]
Hi. I took the files along with the model from the link https://huggingface.co/Systran/faster-whisper-large-v3/tree/main - vocab, tokenizers and others. 
I changed the folder noScribe.app/Contents/Resources/models/faster-whisper-large-v2 without renaming it. Everything works. 
Subjectively, it seems to me that the quality of the translation has become a little better and a little faster. 
I noticed that there are builds for specific languages, they are smaller, lighter and work faster. It would be great to work with a specific language model.

**Reply by kaixxx** [reply]
Thank you for testing. I am surprised that it worked so easy. It seems that the MacOS-Version is already using the newest faster-whisper without any issues. This gives me confidence to update on Windows also. 

> It would be great to work with a specific language model.

You can use the same technique as above to install a language specific model (e.g. replacing the smaller "fast" model). You must, however, use models in the specific format of faster-whisper. If you don't find them in this format, there is also a conversion tool on the GitHub page of faster-whisper.

**Reply by gernophil** [reply]
The fact that it works in macOS might also be because we are quite progressive with packages in the macOS version whereas we are more conservative with those in the windows version. Would be interesting to see, if this also works with the windows version.
